### PR TITLE
Add ServiceMonitor capability for Prometheus Operator

### PR DESCRIPTION
**Pull request by rebelthor**
Prometheus operator changes the way metrics are discovered by not using annotations and rather creating a Custom Resource of ServiceMonitor type: https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources

This merge request adds capability to create a ServiceMonitor custom Resource so that when enabled, Prometheus Operator will automatically discover metrics exposed by kube-eagle

### CODE REVIEW

File: kube-eagle/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.serviceMonitor.create }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app: {{ template "kube-eagle.name" . }}
+    chart: {{ template "kube-eagle.chart" . }}
+    release: prometheus-operator
```
Comment (weeco) [inline kube-eagle/templates/servicemonitor.yaml]: This should be `{{ .Release.Name }}` shouldn't it?

**Review by weeco — CHANGES_REQUESTED** [review]
Thanks for your pull request, looks good :). I believe the metadata label release should be the chart's Release.Name

File: kube-eagle/templates/servicemonitor.yaml (hunk 1 above)
Comment (rebelthor) [reply kube-eagle/templates/servicemonitor.yaml]: The problem with the release label is that the Prometheus Operator [uses a label selector to decide which ServiceMonitors to check](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#include-servicemonitors), in the idea that you may have several prometheuses installed and this provides a way to assign ServiceMonitors to specific Prometheuses.

The default chart installation of Prometheus Operator comes with this configuration:
`prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues = true
prometheus.prometheusSpec.serviceMonitorSelector = {}`

Due to this and the prometheus operator chart template in https://github.com/helm/charts/blob/master/stable/prometheus-operator/templates/prometheus/prometheus.yaml#L69-L75 , the selector ends up being actually the prometheus operator release name:

`  Service Monitor Selector:
    Match Labels:
      Release:  prometheus-operator`

To improve this, I will update this merge request shortly by parameterizing the release label (with a default of prometheus-operator which seems like a sensible default, and also with an additional label value to support custom prometheus operator Match Labels, in which case the release would then be allowed to be {{ .Release.Name }}.

Comment (rebelthor) [reply kube-eagle/templates/servicemonitor.yaml]: I've submitted a new commit which addresses this aspect. Thanks!

**Comment by weeco** [comment]
Merged! Thanks a lot for your effort @rebelthor, I just repackaged it (updated repo index.yaml was missing).

**Pull request merged**